### PR TITLE
Return empty string in Option::get_name() for hidden options

### LINKED ISSUE #332: Hidden option is displayed with (no|incorrect) args
Love your library! I noticed a small issue. I added three flags and made one of them hidden. 

Code Snippet:
```
app.add_flag("--flagA", "flag A");
app.add_flag("--flagB", "flag B");
app.add_flag("--flagC", "flag C")->group("");
app.require_option(1, 1); // Require exactly one flag
```

The hidden flag is not displayed in help message. When I specify no or incorrect arguments, it prints the hidden flag. 

```
# myutil
Exactly 1 option from [--flagA,--flagB,--flagC] is required
Run with --help for more information.

# myutil --fla
Exactly 1 option from [--flagA,--flagB,--flagC] is required
Run with --help for more information.

# myutil --help

[Exactly 1 of the following options is required]
Usage: myutil [OPTIONS]

Options:
  -h,--help            Print this help message and exit
  --flagA              flag A
  --flagB              flag B
```

Please let me know if I am using the library wrong. Thanks for your help.

### PR DESCRIPTION
**Pull request by skannan89**
Fixes https://github.com/CLIUtils/CLI11/issues/332

### CODE REVIEW

File: include/CLI/Option.hpp [hunk 1]
```diff
@@ -627,6 +627,8 @@ class Option : public OptionBase<Option> {
     std::string get_name(bool positional = false, //<[input] Show the positional name
                          bool all_options = false //<[input] Show every option
                          ) const {
+        if (detail::to_lower(get_group()).empty())
```
Comment (skannan89) [inline include/CLI/Option.hpp]: Don't think we need `detail::to_lower()` call.

Comment (phlptp) [reply include/CLI/Option.hpp]: wouldn't appear so if we are just checking empty()

**Comment by henryiii** [comment]
Please rebase and force-with-lease, please. CI should be fixed. You might need to run pre-commit, but unlikely.

I'll do it if you need me to.

**Comment by skannan89** [comment]
Thanks for fixing CI. Added a few more commits.

**Comment by henryiii** [comment]
Thanks!

**Pull request merged**